erts/emulator/beam/erl_bits.c

```diff
@@ -40,6 +40,209 @@
 
 #define BIT_IS_MACHINE_ENDIAN(x) (((x)&BSF_LITTLE) == BIT_ENDIAN_MACHINE)
 
+#if (SIZEOF__FLOAT16 == 2)
+typedef _Float16 erlfp16;
+
+#define FP16_FROM_FP64(x) ((_Float16) x)
+#define FP16_TO_FP64(x) ((double) x)
+#else
+typedef uint16_t erlfp16;
+
+#define FP16_FROM_FP64(x) (fp16_ieee_from_fp32_value((float) x))
+#define FP16_TO_FP64(x) ((double) fp16_ieee_to_fp32_value(x))
+
+/*
+ * C-implementation of Float16 - https://github.com/Maratyszcza/FP16
+ *
+ * The MIT License (MIT)
```

Comment:
This notice needs to be added in `system/COPYRIGHT` as well.

Reply:
Will do. Btw, how would you handle the separate file? Would it be a erl_bits_fp16.h that you just include at the top?

Reply:
Yep.